spec/requests/api/admin/editor_can_publish_article_spec.rb

```diff
@@ -0,0 +1,118 @@
+RSpec.describe 'Api::Admin::Articles :update', type: :request do
+  let!(:article) { create(:article) }
+  let(:editor) { create(:user, role: 'editor') }
+  let(:editors_credentials) { editor.create_new_auth_token }
+  let(:editors_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(editors_credentials) }
+
+  let(:journalist) { create(:user, email: "asd@asd.com", role: 'journalist') }
+  let(:journalist_credentials) { journalist.create_new_auth_token }
+  let(:journalist_headers) { { HTTP_ACCEPT: 'application/json' }.merge!(journalist_credentials) }
+
+  describe 'editor successfully updates and' do
+    before do
+      put "/api/admin/articles/#{article.id}", headers: editors_headers, params: { activity: "PUBLISH", premium: true , category: 'economy'}
+      article.reload()
+    end
+
+    it 'gives a success code' do
+      expect(response).to have_http_status 200
+    end
+
+    it 'gives a success message' do
+      expect(response_json['message']).to eq "Article successfully published!"
+    end
+
+    it 'has set published to true' do
+      expect(article.published).to eq true
+    end
+
+    it 'has set category to economy' do
+      expect(article.category).to eq "economy"
+    end
+
+    it 'has set premium to true' do
+      expect(article.premium).to eq true
+    end
+  end
+
+  describe 'editor successfully updates only few params' do
+    before do
+      put "/api/admin/articles/#{article.id}", headers: editors_headers, params: { activity: "PUBLISH", category: 'economy'}
+      article.reload()
+    end
+
+    it 'gives success status' do
+      expect(response).to have_http_status 200
+    end
+
+    it 'updates the provided params' do
+      expect(article.category).to eq 'economy'
+    end
+
+    it 'but not the not provided ones' do
+      expect(article.premium).to eq false
+    end
+  end
+
+  describe 'article with id does not exist' do
```

Comment:
This is a bit confusing description, maybe could say 'when there is no article with the given id' or something like that?

Reply:
Slightly rephrased